gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -234,6 +237,32 @@ private boolean isDirectPathEnabled(String serviceAddress) {
     return false;
   }
 
+  // DirectPath should only be used on Compute Engine. Follow this doc:
+  // https://docs.google.com/document/d/1xQXE27x9wTvwPsgiX9Hn0o8mcq5z3SKi-1jwscQsCAk/edit#.
+  // Notice Windows will not be supported until we setup tests.
+  private boolean isOnComputeEngine() {
+    String osName = System.getProperty("os.name");
+    if ("Linux".equals(osName)) {
+      String cmd = "cat /sys/class/dmi/id/bios_date && cat /sys/class/dmi/id/bios_vendor";
+      try {
+        Process process = Runtime.getRuntime().exec(new String[] {"/bin/sh", "-c", cmd});
+        process.waitFor();
+        ByteArrayOutputStream resultOutStream = new ByteArrayOutputStream();
+        InputStream processInStream = new BufferedInputStream(process.getInputStream());
+        int num = 0;
+        byte[] bs = new byte[1024];
+        while ((num = processInStream.read(bs)) != -1) {
+          resultOutStream.write(bs, 0, num);
+        }
+        String result = new String(resultOutStream.toByteArray());
+        return result.contains("01/01/2011") && result.contains("Google");
```

Comment:
why this date?

Reply:
See this doc: https://docs.google.com/document/d/1xQXE27x9wTvwPsgiX9Hn0o8mcq5z3SKi-1jwscQsCAk/edit#heading=h.7ioa2p9bw6l1

Reply:
As this is open source code, can this be made with a reference to a public doc? 
Also this needs a comment in the source code explaining it, and possibly a named constant for the date.

Reply:
Sorry I could not find a public doc, so I just removed the link in the code. Two constants are added though.

Reply:
per doc, Virtual machines started after August 2016 have access to a more reliable "product name" string which is equal to "Google Compute Engine".

Reply:
"Google Compute Engine" is for /sys/class/dmi/id/product_name, but for now we are checking /sys/class/dmi/id/bios_date and /sys/class/dmi/id/bios_vendor in the code. Do you think we also need to check /sys/class/dmi/id/product_name?

Reply:
I'm just going by the doc you cited. I read it as saying one should use product name instead of bios id and vendor

Reply:
I see, yeah, thanks, product name is used now.